chat-client/src/client/chat.ts

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable prefer-const */
+import { TabEventParams } from '@aws/language-server-runtimes/protocol'
+import { CHAT_PROMPT, NEW_TAB_CREATED, ServerMessage, TAB_CHANGED, TAB_REMOVED } from '../contracts/serverContracts'
+import {
+    AUTH_NEEDED_EXCEPTION,
+    ERROR_MESSAGE,
+    SEND_TO_PROMPT,
+    SendToPromptMessage,
+    TAB_ID_RECEIVED,
+    TabIdReceivedParams,
+    UI_FOCUS,
+    UI_IS_READY,
+    UiMessage,
+} from '../contracts/uiContracts'
+import { Messager, OutboundChatApi } from './messager'
+import { InboundChatApi, createMynahUI } from './mynahUi'
+import { TabFactory } from './tabs/tabFactory'
+import { TabStorage } from './tabs/tabStorage'
+
+export const createChat = (clientApi: { postMessage: (msg: UiMessage | ServerMessage) => void }) => {
```

Comment:
Not limited to this file, but can we add unit tests? does it make sense to have any for the chat client?

Reply:
Yes, will do it as a followup.